numpy/core/shape_base.py

```diff
@@ -447,8 +447,12 @@ def block_recursion(self, arrays, depth=0):
             # We've 'bottomed out' - arrays is either a scalar or an array
             # type(arrays) is not list
             return atleast_nd(arrays, result_ndim)
-
-    return block_recursion(arrays)
+    if max_depth == 0:
+        # Catch an edge case where the user provies a single array to
+        # block. block should return a copy of that array and not a view.
+        return arrays.copy()
```

Comment:
For consistency with the other code path, it probably makes sense to call `block_recursion(arrays)` anyway

Reply:
so `return block_recursion(arrays.copy())`

Reply:
`block_recursion(arrays).copy()` would be my choice

Reply:
ok.